homeassistant/components/workday/config_flow.py

```diff
@@ -0,0 +1,259 @@
+
+    data: dict[str, Any] = {}
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(
+        config_entry: ConfigEntry,
+    ) -> WorkdayOptionsFlowHandler:
+        """Get the options flow for this handler."""
+        return WorkdayOptionsFlowHandler(config_entry)
+
+    async def async_step_import(self, config: dict[str, Any] | None) -> FlowResult:
+        """Import a configuration from config.yaml."""
+
+        self._async_abort_entries_match(config)
+        return await self.async_step_options(user_input=config)
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the user initial step."""
+        if user_input is not None:
+            self.data = user_input
+            return await self.async_step_options()
+        return self.async_show_form(
+            step_id="user",
+            data_schema=DATA_SCHEMA_SETUP,
+            errors={},
+        )
+
+    async def async_step_options(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle remaining flow."""
+        errors = {}
+        if user_input is not None:
+            try:
+                await self.hass.async_add_executor_job(
+                    validate_custom_dates, {**self.data, **user_input}
+                )
+            except AddDatesError:
+                errors["add_holidays"] = "add_holiday_error"
+            except RemoveDatesError:
+                errors["remove_holidays"] = "remove_holiday_error"
+            except NotImplementedError:
+                self.async_abort(reason="incorrect_province")
+
+            if not errors:
+                name = self.data.get(CONF_NAME, user_input.get(CONF_NAME))
+                return self.async_create_entry(
+                    title=name,
+                    data={},
+                    options={**self.data, **user_input},
+                )
+
+        new_schema = await self.hass.async_add_executor_job(
+            get_providence_to_schema, DATA_SCHEMA_OPT, self.data
+        )
+        return self.async_show_form(
+            step_id="options",
+            data_schema=new_schema,
+            errors=errors,
+        )
+
+
+class WorkdayOptionsFlowHandler(OptionsFlowWithConfigEntry):
+    """Handle Workday options."""
+
+    async def async_step_init(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Manage Workday options."""
+        errors = {}
+
+        if user_input is not None:
+            try:
+                await self.hass.async_add_executor_job(
+                    validate_custom_dates, {**self.options, **user_input}
+                )
+            except AddDatesError:
+                errors["add_holidays"] = "add_holiday_error"
+            except RemoveDatesError:
+                errors["remove_holidays"] = "remove_holiday_error"
+            else:
+                return self.async_create_entry(
```

Comment:
Do we want to implement some kind of "cancel on matching dict"?
There was a discussion about it in https://github.com/home-assistant/core/pull/89914